### PR TITLE
Fix setEnabled for GridPro

### LINKED ISSUE #25: setEnabled(false) does not disable editing
GridPro implements HasEnabled interface, and hence has setEnabled(..) and isEnabled() methods. setEnabled(false) however seems not to have effect on editing. I.e. editing is possible even when setEnabled(false)

Use the following example to reproduce

```
import com.vaadin.flow.component.gridpro.GridPro;
import com.vaadin.flow.component.orderedlayout.FlexLayout;
import com.vaadin.flow.router.*;
import com.vaadin.flow.server.PWA;

@Route("")
@PWA(name = "Project Base for Vaadin Flow with CDI", shortName = "Project Base")
@PageTitle("Pinnacle Professional")
@SuppressWarnings("unused" /*main entry point!*/)
public class MainView extends FlexLayout implements RouterLayout {

    public MainView() {
        GridPro<String[]> minimalGrid = new GridPro<>();
        minimalGrid.addEditColumn((ar) -> ar[0])
            .text((ar, s) -> ar[0] =s);
        minimalGrid.setEnabled(false);
        minimalGrid.setItems(new String[]{"abc"}, new String[]{"def"});
        add(minimalGrid);
    }
}
```

### PR DESCRIPTION
**Pull request by shadikhani**
Fix #25

### CODE REVIEW

File: vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java [hunk 1]
```diff
@@ -108,6 +108,12 @@ public GridPro(int pageSize) {
         setup();
     }
 
+    @Override
+    public void onEnabledStateChanged(boolean enabled) {
+        super.onEnabledStateChanged(enabled);
+        getElement().executeJs("this._editingDisabled = $0;", !enabled);
```
Comment (tomivirkki) [inline vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java]: Why use `executeJs` for this? I assume the grid wouldn't have `_editingDisabled` on anymore after page refresh (or grid hide/unhide) if `executeJs` is used.

Comment (shadikhani) [reply vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java]: Done.

**Comment by tamasnet** [comment]
What if a cell editor is currently focused when this happens? Ideally the pending edit should be committed similar to if the grid had lost focus or the cell selection changed.

**Comment by web-padawan** [comment]
@tamasnet thanks for pointing out. We can test that separately. Currently `_editingDisabled` flag only prevents a new editor from being open, it doesn't affect currently open editor.

I have submitted a separate issue to the web component: https://github.com/vaadin/vaadin-grid-pro/issues/109

File: vaadin-grid-pro-flow-integration-tests/src/test/java/com/vaadin/flow/component/gridpro/test/BasicIT.java [hunk 2]
```diff
@@ -175,21 +175,26 @@ public void selectEditorOptionsAreSet() {
         Assert.assertTrue(optionsList.contains("Sales"));
     }
 
+    @Test
+    public void disabledGridShouldNotBeActivatedByDoubleClick() {
```
Comment (tomivirkki) [inline vaadin-grid-pro-flow-integration-tests/src/test/java/com/vaadin/flow/component/gridpro/test/BasicIT.java]: The test isn't valid. It passes even without the change in GridPro.java

Comment (shadikhani) [reply vaadin-grid-pro-flow-integration-tests/src/test/java/com/vaadin/flow/component/gridpro/test/BasicIT.java]: Done.

**Pull request merged**